bit_manipulation/binary_exponentiation.cpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ * @brief Binary exponentiation for calculating power
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂

Reply:
Sure !